brightwind/transform/transform.py

```diff
@@ -1270,89 +1270,119 @@ def offset_timestamps(data, offset, date_from=None, date_to=None, overwrite=Fals
+        # To decrease 10 minutes within a given date range not overwriting the original data
+        op2 = bw.offset_timestamps(data, offset='-10min', date_from='2016-02-01 00:20:00',
+            date_to='2016-02-01 01:40:00')
 
-        #To decrease 10 minutes within a given date range not overwriting the original data
-        op2 = bw.offset_timestamps(data, offset='-10min', date_from='2016-01-01 00:20:00',
-            date_to='2016-01-01 01:40:00')
+        # To decrease 30 minutes within a given date range not overwriting the original data and giving as input dates
+        # for date_from and date_to
+        op2 = bw.offset_timestamps(DATA, offset='-30min', date_from='2016-01-09', date_to='2016-01-10')
 
-        #Can accept Series or index as input
-        op3 = bw.offset_timestamps(data.Spd80mS, offset='1D', date_from='2016-01-01 00:20:00')
+        # Can accept Series or index as input
+        op3 = bw.offset_timestamps(data.Spd80mS, offset='1D', date_from='2016-02-01 00:20:00')
 
-        op4 = bw.offset_timestamps(data.index, offset='-10min', date_from='2016-01-01 00:20:00',
-            date_from='2016-01-01 01:40:00')
+        op4 = bw.offset_timestamps(data.index, offset='-10min', date_from='2016-02-01 00:20:00',
+            date_to='2016-02-01 01:40:00')
 
-        #Can also except decimal values for offset, like 3.5H for 3 hours and 30 minutes
+        # Can also except decimal values for offset, like 3.5H for 3 hours and 30 minutes
+        op5 = bw.offset_timestamps(data.index, offset='3.5H', date_from='2016-02-01 00:20:00',
+            date_to='2016-02-01 01:40:00')
 
-        op5 = bw.offset_timestamps(data.index, offset='3.5H', date_from='2016-01-01 00:20:00',
-            date_from='2016-01-01 01:40:00')
+        # Can accept also Timestamp and datetime objects
+        bw.offset_timestamps(data.index[0], offset='4H')
+        bw.offset_timestamps(datetime.datetime(2016, 2, 1, 0, 20), offset='3.5H')
+        bw.offset_timestamps(datetime.date(2016, 2, 1), offset='-5H')
+        bw.offset_timestamps(datetime.time(0, 20), offset='30min')
 
     """
-    
-    date_from = pd.to_datetime(date_from)
-    date_to = pd.to_datetime(date_to)
 
-    if isinstance(data, pd.Timestamp) or isinstance(data, datetime.date)\
-            or isinstance(data, datetime.time)\
-            or isinstance(data, datetime.datetime):
+    if pd.isnull(date_from):
+        if isinstance(data, pd.DatetimeIndex):
+            date_from = data[0]
+        elif isinstance(data, pd.Series) or isinstance(data, pd.DataFrame):
+            date_from = data.index[0]
+    else:
+        date_from = pd.to_datetime(date_from)
+
+    if pd.isnull(date_to):
+        if isinstance(data, pd.DatetimeIndex):
+            date_to = data[-1]
+        elif isinstance(data, pd.Series) or isinstance(data, pd.DataFrame):
+            date_to = data.index[-1]
+    else:
+        date_to = pd.to_datetime(date_to) - datetime.timedelta(seconds=1)
```

Comment:
Hi @BiancaMorandi ,
Can you remember why we take 1 sec off here but not in other places when we go to less than the date_to?

Reply:
@stephenholleran as requested I have updated `offset_timestamps` and `slice_data` functions to use less than for date_to instead of taking 1 sec off.